borsh/src/schema/container_ext/validate.rs

```diff
@@ -92,13 +107,9 @@ fn validate_impl<'a>(
             tag_width,
             variants,
         } => {
-            // TODO(#230): variants.len() is actually incorrect here.  Since
-            // arbitrary discriminant can be used for variants, an actual
-            // largest discriminant may be larger than the number of variants.
-            // For the time being we’re using variants.len() as a proxy but
-            // eventually variant discriminants should be added to the schema.
-            let max_discriminant = variants.len().saturating_sub(1) as u64;
-            check_tag_width(declaration, *tag_width, max_discriminant)?;
+            if *tag_width > U64_LEN {
+                return Err(Error::TagTooWide(declaration.to_string()));
+            }
```

Comment:
So `length_width: 3` is invalid (for whatever reason), but `tag_width: 3` is valid?